### ISSUE TITLE
add place does not work on android-4.4

### ISSUE THREAD

**Issue by k3b** [gabm/FancyPlaces#2]
First of all: i think it is a fantastic idea to have an app that can remember name+(gps-)place+datetime+notes.

the current version downloaded from fdriod (build-date 2015-06-03) seems to not work on my device

when i push the add button "+" i get the addPlaceActivity but i have found no way to store a recored.location:

There is neither a "save button" nor any status/errormessage indicating that the location cannot be saved. Using the "back-button" seems to cancel the location recording.

the map in the addPlaceActivity does not show my gps location although i have a gps-fix (checked with the SatStat-app)

I am using a android-4.4 smartphone with no "google-play-services" installed and therefore "corse-internet-location-detection" that detects "location via known wlan-networks" does not work on my device.

If you you need support with the map-display via osmdroid maybe i can help you

**Comment by gabm** [comment]
Thank you for your bug report! 

As already seen in Issue #1, there seem to be trouble on devices <=4.4 - I'm currently investigating...

The app should check both - Wifi and GPS for whatever comes first ...both location provider should be accepted. 

The general usage should be as follows:
- press the '+' button
- put a title
- get a location found (automatically or by pressing "Update Location")
  - the found location will be marked in the map
- press the "check"-sign on the right side of the title bar to save!

There should be at least a toast indicating when for some reason it cant save (e.g. no location)... i will implement this for next version!

**Comment by k3b** [comment]
I have checked it again with the app binary from ticket #1.

The problem seems that i cannot save until there is a locaton fix.

On my android 4.2 tablet with installed google-playservices and with no gps:
 the app works as expected as long as there is a wifi-internetconnection. In this case  the "corarse location" is queried by google-playserves via wlan-lookup via internet.

if i disconnect wlan and kill FancyPlaces and restart it there is no know locaton and therefore saving is not possible any more.

coarse location does not work without google.

On my android 4.4 cellfone (that has no google-playservices) i donot get a location fix although gps and internet are on and therefor i cannot save.

I assmume that there is some problem getting the "fine location".

```
  > There should be at least a toast indicating when for some 
  > reason it cant save (e.g. no location)..
```

This is a good idea

**Comment by gabm** [comment]
thanks for your big investigation :) it's very helpful!

1) there will be a toast whenever a fancy place can't be stored, if either title or location are empty... or both

2) i will have a deeper look in gps-only localization... the absence of play services shouldn't play any role here, right?

Thanks again! there will probably be a bugfix release with this and the other bugfix

**Comment by jojorb** [comment]
indeed, i'll follow here. thx

**Comment by gabm** [comment]
Hey k3b,

i remember that i have been struggling writing a location handler that properly uses all providers available and gets suspended and restored properly during the activities life time...

Obviously the problem lies in the way I retrieve the users location...I saw that u are also working on several projects of localisation/osmdroid/osm etc.. could u point me to a piece of code which u can use successfully on your 4.4.4 device to retrieve the users location without play services from gps?

That would be very helpful!

**Comment by gabm** [comment]
Hey guys, 

i think i found the problem - a combination of several small mistakes... although i think i should rework the way i retrieve locations, for now this should fix your error.

Would you please check [this](https://www-user.tu-chemnitz.de/~gabm/FancyPlaces/FancyPlaces.apk) Version and report back?

**Comment by jojorb** [comment]
Hi thanks for this but when i did try to add a new test file tittle is ok, photos is ok but update location when push return "updating location" and nothing, Notes is ok, when trying to save by clicking the big tick it return: "error saving, check if you provide a valid title and localisation!"

**Comment by gabm** [comment]
I'm very sorry to hear that.. Finally i set up my phone without play services and i was able to reproduce some things...

Could u please confirm that:
1) Gps location is working in other apps
2) the location symbol in the status bar starts to pop up when u hit "update location"and disappears after a short while (<2min)
3) at some point it works when u hit " update location" for several times?

Currently I'm reworking the code of how i retrieve users location, there seem to be additional issues...

I'm sorry for inconvenience!

**Comment by gabm** [comment]
hey guys, well i reworked the LocationHandler which is now a single static object to reflect that the users location is one single static state...

I cannot reproduce any problems with my phone that has no play services anymore... Please redownload [this](https://www-user.tu-chemnitz.de/~gabm/FancyPlaces/FancyPlaces.apk) and report back.. hopefully for the last time :)

Thank you a lot for your time!

**Comment by jojorb** [comment]
thank you, so for 
1- yes and no, by time to time if i can say, (i sometime need a reboot to make it work)
it could be helpful to exactly know witch localisation app to install to have it flawless ex: LocalWifiNIlBackend ...
2- yes it popup well 
3- it did not most of the time, but it could
but when it work you app is very good and user friendly

on the previous version i manage to make it work but i don't know how i did that, i mean i install so many pkg.. and different setup with localisation control.

on your last version it works well actually, but as i mention earlier it could be nice to have a kind of guide for non GAPP user and witch pkg to you may recommend   for localisation.

thank you

**Comment by gabm** [comment]
Thank you for your input! 

Nice that it seems to work now - actually the builds you tested are just test builds that help me to find out, where the issues may lie ... then i made some debug builds for you to test and to see if the problem is finally fixed... as soon as you give me a positive feedback that the problem is fixed i will push the fixes to the f-droid store so that there will be only one single most recent version that is functional for all users...

Thanks a lot for your help - i consider the issue to be solved - the next version will be v1.1.0 wich contains the fix for your problem!

**Comment by k3b** [comment]
I have verified that the version from ticket #7 now works on my non google android device.

Thanks for fixing it
